AppCenterDataStorage/AppCenterDataStorageTests/MSTokenTests.m

```diff
@@ -21,6 +37,199 @@ - (void)tearDown {
+
+  // Then
+  XCTAssertTrue([partitionName isEqualToString:resultDic[kMSPartition]]);
+  XCTAssertTrue([token isEqualToString:resultDic[kMSToken]]);
+  XCTAssertTrue([status isEqualToString:resultDic[kMSStatus]]);
+  XCTAssertTrue([dbName isEqualToString:resultDic[kMSDbName]]);
+  XCTAssertTrue([dbAccount isEqualToString:resultDic[kMSDbAccount]]);
+  XCTAssertTrue([dbCollectionName isEqualToString:resultDic[kMSDbCollectionName]]);
+  XCTAssertTrue([expiresOn isEqualToString:resultDic[kMSExpiresOn]]);
+}
+
+- (void)testGetTokenResultWithString {
+
+  // If
+  NSString *tokenString =
+      [NSString stringWithFormat:@"{\"%@\":\"%@\",\"%@\":\"%@\",\"%@\":\"%@\",\"%@\":\"%@\",\"%@\":\"%@\",\"%@\":\"%@\",\"%@\":\"%@\"}",
+                                 kMSPartition, partitionName, kMSToken, token, kMSStatus, status, kMSDbName, dbName, kMSDbCollectionName,
+                                 dbCollectionName, kMSExpiresOn, expiresOn, kMSDbAccount, dbAccount];
+
+  // When
+  MSTokenResult *result = [[MSTokenResult alloc] initWithString:tokenString];
+
+  // Then
+  XCTAssertTrue([partitionName isEqualToString:result.partition]);
+  XCTAssertTrue([token isEqualToString:result.token]);
+  XCTAssertTrue([status isEqualToString:result.status]);
+  XCTAssertTrue([dbName isEqualToString:result.dbName]);
+  XCTAssertTrue([dbCollectionName isEqualToString:result.dbCollectionName]);
+  XCTAssertTrue([expiresOn isEqualToString:result.expiresOn]);
+  XCTAssertTrue([dbAccount isEqualToString:result.dbAccount]);
+
+  // When
+  NSString *resultString = [result serializeToString];
+
+  NSDictionary *resultDic = [self deserializeDataString:resultString];
+  NSDictionary *tokenDic = [self deserializeDataString:tokenString];
+
+  // Then
+  XCTAssertTrue([resultDic isEqualToDictionary:tokenDic]);
+}
+
+- (void)testGetTokenResultWithDictionary {
+
+  // If
+  NSMutableDictionary *tokenDic = [@{
+    kMSPartition : partitionName,
+    kMSToken : token,
+    kMSStatus : status,
+    kMSDbName : dbName,
+    kMSDbAccount : dbAccount,
+    kMSDbCollectionName : dbCollectionName,
+    kMSExpiresOn : expiresOn
+  } mutableCopy];
+
+  // When
+  MSTokenResult *result = [[MSTokenResult alloc] initWithDictionary:tokenDic];
+
+  // Then
+  XCTAssertEqual(tokenDic[kMSPartition], result.partition);
+  XCTAssertEqual(tokenDic[kMSToken], result.token);
+  XCTAssertEqual(tokenDic[kMSStatus], result.status);
+  XCTAssertEqual(tokenDic[kMSDbName], result.dbName);
+  XCTAssertEqual(tokenDic[kMSDbAccount], result.dbAccount);
+  XCTAssertEqual(tokenDic[kMSDbCollectionName], result.dbCollectionName);
+  XCTAssertEqual(tokenDic[kMSExpiresOn], result.expiresOn);
+
+  // When
+  NSString *resultString = [result serializeToString];
+  NSDictionary *resultDic = [self deserializeDataString:resultString];
+
+  // Then
+  XCTAssertTrue([resultDic isEqualToDictionary:tokenDic]);
+}
+
+- (void)testGetTokenResponseWithTokenList {
+
+  // If
+  NSMutableDictionary *tokenDic1 = [@{
```

Comment:
This code (lines 140-159) is duplicated with the method below, perhaps refactor into a helper method?